### PR TITLE
First attempt and logging and OOP changes.

### PR DESCRIPTION
**Pull request by hahnd**
I changed the logging over and switched the model processes to classes.  I am not sure how to test this code, but I think you both know better than I do.  Please let me know if anything is unclear -- I want to make sure everyone is comfortable with these changes.

- [ ] Do these changes modify the system output in any way? **[Yes or No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [ ] Describe testing already performed for these changes:</br>

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes or No]**

- [ ] Do these changes include sufficient testing updates? **[Yes or No]**

- [ ] Will this PR result in changes to the test suite? **[Yes or No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
- [ ] Review the source issue metadata (labels, project, and milestone).
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Project**
Select: **Milestone** as the version that will include these changes
Select: **Development** to link to the original development issue.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Review by mkavulich — CHANGES_REQUESTED** [review]
@hahnd There is still a problem running the test case, 

```
2022-08-30 20:42:09.000 +0000 - DEBUG - wrfcloud-test - Ungrib - Executing ungrib.exe
Traceback (most recent call last):
  File "/glade/scratch/kavulich/wrfcloud/PR_60/wrfcloud/python/src/wrfcloud/runtime/./run.py", line 74, in <module>
    main()
  File "/glade/scratch/kavulich/wrfcloud/PR_60/wrfcloud/python/src/wrfcloud/runtime/./run.py", line 50, in main
    log.debug(ungrib.get_run_summary())
  File "/glade/scratch/kavulich/wrfcloud/PR_60/wrfcloud/python/src/wrfcloud/runtime/__init__.py", line 91, in get_run_summary
    return f'{self.__class__} {self.success:"succeeded":"failed"} in {self.get_elapsed_time()} seconds.'
ValueError: Invalid format specifier
```

I don't understand the `self.success:"succeeded":"failed"` syntax; is this correct and just not being parsed correctly? Maybe my version of python isn't new enough? (I'm testing with 3.9.13)

**Comment by hahnd** [comment]
@mkavulich - Yes, this code is failing.  I just pushed a fix for it.

This was supposed to be a [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) (although I got that syntax wrong too).  Looks like Python does not support the standard syntax and had to do it a bit differently.

**Comment by hahnd** [comment]
Just fixed the merge conflict in setup.py too.

**Review by mkavulich — APPROVED** [review]
Looks good after the latest changes; test on Cheyenne was successful

**Comment by fossell** [comment]
@mkavulich - did you have to make a mod locally for what logging error you saw before? Does that need to be pushed to this branch too to make this work?

File: python/src/wrfcloud/runtime/run.py [hunk 1]
```diff
@@ -9,84 +9,65 @@
...
-    logging.debug('Creating new RunInfo')
+    log.info(f'Starting new run "{name}"')
+    log.debug('Creating new RunInfo')
     runinfo = RunInfo(name)
-    logging.info(f'Setting up working directory {runinfo.wd}')
-    setup_logging(logdir=runinfo.wd, logfile='debug.log')
-    logging.debug(f'Moving setup.log to {runinfo.wd}')
+    log.info(f'Setting up working directory {runinfo.wd}')
+    log.debug(f'Moving setup.log to {runinfo.wd}')
     os.rename('setup.log', runinfo.wd + '/setup.log')
```
Comment (mkavulich) [inline python/src/wrfcloud/runtime/run.py:42]:
```suggestion
```

**Review by mkavulich — CHANGES_REQUESTED** [review]
Oh shoot, yes, I forgot I made that local mod. @hahnd it's a simple fix; it will need some cleanup that I can apply in a future PR but this quick fix will ensure it works.

**Comment by fossell** [comment]
@mkavulich, @hahnd - Thanks! Now it works on Cheyenne for a quick test.  One thing I noticed, if the 'workdir' is set to a non-existent directory, the program fails. If it's set to a directory that already exists, then it is successful. I don't think this was the case before, and I'm not sure what the needed behavior is now given the changes. Just wanted to make note of it.  
For example, in run.yml:
`workdir: /glade/scratch/fossell/WRFcloudFramework/wrfcloud/python/src/wrfcloud/runtime/
`

This is successful because runtime already exists. 

If you wanted to change this path to redirect the output to different directory (as I had it set up from before testing an old PR), I could change this to:
`workdir: /glade/scratch/fossell/WRFcloudFramework/wrfcloud/python/src/wrfcloud/runtime/output`
 
But now, this fails with:

>   File "/glade/scratch/fossell/WRFcloudFramework/wrfcloud/python/src/wrfcloud/runtime/ungrib.py", line 88, in run
>     os.mkdir(self.runinfo.ungribdir)
> FileNotFoundError: [Errno 2] No such file or directory: '/glade/scratch/fossell/WRFcloudFramework/wrfcloud/python/src/wrfcloud/runtime/output/ungrib'

Because that directory was not created.  Not sure if it matters, just wanted to point it out!

**Review by fossell — APPROVED** [review]
Approved on my end since it did work. @mkavulich - could you take look at my previous comment and jsut let the team know if we need to address this at all. And then re-approve if you're ok with this PR again. I think that will be sufficient to merge this then.

**Comment by mkavulich** [comment]
@fossel I think that's a problem we can deal with in a future commit; for now let's go ahead and merge

**Review by mkavulich — APPROVED** [review]

**Pull request merged**